rmgpy/molecule/fragment_utils.py

```diff
@@ -310,3 +313,332 @@ def merge_frag_list(to_be_merged):
     # print('{}% of fragments fully merged...'.format(np.round(100*(i+1)/len(flattened_matches_random)),1))
 # print(newfraglist)
     return newfraglist
+
+def check_if_radical_near_cutting_label(species_smiles):
+    species_fragment = Fragment().from_smiles_like_string(species_smiles)
+    for i, atom in enumerate(species_fragment.atoms):
+        species_fragment.atoms[i].id = i
+    for atom in species_fragment.atoms:
+        if atom.radical_electrons == 1:
+            bonded_atoms = list(atom.bonds.keys())
+            bonded_atom_types = [type(x) for x in bonded_atoms]
+            if CuttingLabel in bonded_atom_types:
+                radical_near_cutting_label = atom
+                nearest_cutting_label = bonded_atoms[bonded_atom_types.index(CuttingLabel)]
+                return species_fragment, nearest_cutting_label
+            for atom2 in bonded_atoms:
+                bonded_atoms_2 = list(atom2.bonds.keys())
+                bonded_atom_types_2 = [type(x) for x in bonded_atoms_2]
+                if CuttingLabel in bonded_atom_types_2:
+                    radical_near_cutting_label = atom2
+                    nearest_cutting_label = bonded_atoms_2[bonded_atom_types_2.index(CuttingLabel)]
+                    return species_fragment, nearest_cutting_label
+    else:
+        return False
+
+def check_if_pibond_near_cutting_label(species_smiles):
+    species_fragment = Fragment().from_smiles_like_string(species_smiles)
+    for i, atom in enumerate(species_fragment.atoms):
+        species_fragment.atoms[i].id = i
```

Comment:
Why is this needed?

Reply:
this is needed because we switch between molecule and rdkit during cutting